application/app/src/main/java/com/example/buber/Services/ApplicationService.java

```diff
@@ -21,4 +26,56 @@ public static User createNewUser(
             String phoneNumber
     ) { return null; }
 
+
+//    --------------------MADEEHA---------------------------------------
+    /*TODO:
+       //get username and password from UI when user creates an account
+       //append @Uber.com to the username
+       //call my create account function in LoginAuth:              LoginAuth.createAccount(username@uber.com, password, (method called on sucess))
```

Comment:
This is great, but lets just worry about getting a User object. Next week / sprint we can think a little more about how to do driver/rider stuff. So right now I think we need the service methods. 
1. SignIn/Login
- takes a username and password, calls ur AuthManger login functionality and gets the docId for the user
- calls ur DB.get(docId) for the users collection and returns that user
2. Logout
- takes a username and password, calls ur DB logout function and returns null
3. CreateNewUser
- takes email, id? and creates a new FireBase user using the AuthManger createAccount function
- Return the FireBaseUser docId, create a new User class (using rest of parameters passed down from UI)
- Take new User and stick in User collections using the DB.create user function

Reply:
It is okay if we have lots of null fields in the User, I am literally just concerned about making sure we can log in / log out / create an account. We can edit the User class so it only takes the attributes Mike and Nick defined in the UI.

Reply:
"worry about getting a User object"- okay so its a little bit confusing so I'm going to explain what I think and then u can tell me if I'm wrong? 
The user class has attributes like location and account and all that.
(For now, let's assume its a name and a number)
The driver's class extends the user. Rider class extends the user. (this means the driver and rider inherit attributes of the user. )
so when "create account" is called it takes in an email and password and this is stored in the firebase Auth section. 
So now for the collections:
It does not make sense to create a user collection.
it makes more sense to create a rider or driver with the same DocID and then cast them to a user.

Reply:
I think we are on the same page, but I believe you cannot create multiple documents with the same DocID as the one generated by the FireBaseUser. What I am concerned about is I don't think we all have completely thought over how we want the driver/rider distinction to work so for now (just to get the app auth up and running) I think its better to just have a users collection **for now** and worry about driver rider at a later point after we have given it some thought. It could come down to storing everything in a user collection and casting to Driver/Rider based on how the user is signed in, or have seperate rider/driver collections.

Reply:
If we want just a user for now, with a name we can also just use "currentuser" in my DBAuth class. you can do "getcurrentUser().getEmail()" or "getcurrentUser().setName()" so this is basically a user object we can pass around. Just a thought.